auditbeat/tracing/endian.go

```diff
@@ -0,0 +1,42 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package tracing
+
+import (
+	"encoding/binary"
+	"unsafe"
+)
+
+// MachineEndian is either binary.BigEndian or binary.LittleEndian, depending
+// on the current architecture.
+var MachineEndian = getCPUEndianness()
+
+func getCPUEndianness() binary.ByteOrder {
+	myInt32 := new(uint32)
+	copy((*[4]byte)(unsafe.Pointer(myInt32))[:], []byte{0x12, 0x34, 0x56, 0x78})
+	switch *myInt32 {
+	case 0x12345678:
+		return binary.BigEndian
+	case 0x78563412:
+		return binary.LittleEndian
+	default:
+		panic("cannot determine endianness")
+	}
+}
```

Comment:
Since we are using unsafe for the endianness assessment, is there a reason to not just do unsafe punning?

Reply:
thx for the comment @efd6 . I tried to avoid doing "modernisation" changes but since you suggested them definitely not a prob. So here since beats in main switched to go 1.21 we can use directly the following

```
var MachineEndian = binary.NativeEndian
``` 
and remove completely `getCPUEndianness()` how does that sound to you?

Reply:
SGTM